dynamicprefixvici.c

```diff
@@ -35,29 +45,80 @@ static void ParseCommandLineArguments(int argc, char** argv, CommandLineArgument
  */
 static void Usage();
 
+/**
+ * @brief Converts a command line argument to an integer
+ * Prints an error message if the argument is invalid and exits
+ * 
+ * @param value 
+ * @param optionname 
+ * @return int 
+ */
+static int parse_option_to_integer(char* value, char* optionname);
+
+/**
+ * @brief Checks whether the string only contains the character specified
+ * Returns true if the string only contains the characters
+ * @param string 
+ * @param characters 
+ * @return true 
+ * @return false 
+ */
+static bool check_only_contains(char* string, char* characters);
+
+/**
+ * @brief Check whether something is a valid IPv6 address
+ * 
+ * @param adres 
+ * @return true 
+ * @return false 
+ */
+static bool check_adres(char* adres);
+
+/**
+ * @brief Checks wheters de input arguments are all valid
+ * 
+ * @param argumenten 
+ * @return true 
+ * @return false 
+ */
+static bool check_validity_of_arguments(CommandLineArguments* argumenten);
+
+/**
+ * @brief Formats the output adress to be given to vici
+ * 
+ * @param argumenten 
+ * @param output 
+ */
+void format_output(CommandLineArguments* argumenten, char* output);
+
+
 int main(int argc, char** argv)
 {
     // Parse command line arguments
     CommandLineArguments arguments;
     ParseCommandLineArguments(argc, argv, &arguments);
+    if(!check_validity_of_arguments(&arguments))
```

Comment:
Ik zou CheckValidityOfArguments een void maken en in de functie zelf exiten, gezien daar ook al de error geprint wordt